### ISSUE TITLE
 pool connection destroyed

### ISSUE THREAD

**Issue by tjosm** [cazala/coin-hive-stratum#61]
I'm seeing this on the logs and my miners are stopped.

```
[Nov 24th 04:55] websocket server created
[Nov 24th 04:55] using custom server
[Nov 24th 04:55] new pool connection (46WNbmwXpYxiBpkbHjAgjC65cyzAxtaaBQjcGpAZquhBKw2r8NtPQniEgMJcwFMCZzSBrEJtmPsTR54MoGBDbjTi2W1XmgM)
[Nov 24th 04:55] host: la01.supportxmr.com
[Nov 24th 04:55] port: 3333
[Nov 24th 04:55] pass: donations
[Nov 24th 04:55] new miner connection
[Nov 24th 04:55] message from miner (unauthenticated) {"type":"auth","params":{"site_key":"46WNbmwXpYxiBpkbHjAgjC65cyzAxtaaBQjcGpAZquhBKw2r8NtPQniEgMJcwFMCZzSBrEJtmPsTR54MoGBDbjTi2W1XmgM","type":"anonymous","user":null,"goal":0}}
[Nov 24th 04:55] message sent to pool (46WNbmwXpYxiBpkbHjAgjC65cyzAxtaaBQjcGpAZquhBKw2r8NtPQniEgMJcwFMCZzSBrEJtmPsTR54MoGBDbjTi2W1XmgM): {"id":1,"method":"login","params":{"login":"46WNbmwXpYxiBpkbHjAgjC65cyzAxtaaBQjcGpAZquhBKw2r8NtPQniEgMJcwFMCZzSBrEJtmPsTR54MoGBDbjTi2W1XmgM","pass":"donations"}}
[Nov 24th 04:55] new miner connection
[Nov 24th 04:55] message from miner (unauthenticated) {"type":"auth","params":{"site_key":"46WNbmwXpYxiBpkbHjAgjC65cyzAxtaaBQjcGpAZquhBKw2r8NtPQniEgMJcwFMCZzSBrEJtmPsTR54MoGBDbjTi2W1XmgM","type":"anonymous","user":null,"goal":0}}
[Nov 24th 04:55] message sent to pool (46WNbmwXpYxiBpkbHjAgjC65cyzAxtaaBQjcGpAZquhBKw2r8NtPQniEgMJcwFMCZzSBrEJtmPsTR54MoGBDbjTi2W1XmgM): {"id":2,"method":"login","params":{"login":"46WNbmwXpYxiBpkbHjAgjC65cyzAxtaaBQjcGpAZquhBKw2r8NtPQniEgMJcwFMCZzSBrEJtmPsTR54MoGBDbjTi2W1XmgM","pass":"donations"}}
[Nov 24th 04:55] new pool connection (4A5Zv5vpNcG4Tk7pF6wRHha6jZEgoF9UgFv49qijdHxeLSLYCWyCbHZA6sqBiWRSwcBcnBrJuEWxjYgtwJAQnzcT3BDvq8m)
[Nov 24th 04:55] host: pool.supportxmr.com
[Nov 24th 04:55] port: 3333
[Nov 24th 04:55] pass: x
[Nov 24th 04:55] message from pool (46WNbmwXpYxiBpkbHjAgjC65cyzAxtaaBQjcGpAZquhBKw2r8NtPQniEgMJcwFMCZzSBrEJtmPsTR54MoGBDbjTi2W1XmgM): {"id":1,"jsonrpc":"2.0","error":{"code":-1,"message":"No active block template"}}
[Nov 24th 04:55] pool connection destroyed (46WNbmwXpYxiBpkbHjAgjC65cyzAxtaaBQjcGpAZquhBKw2r8NtPQniEgMJcwFMCZzSBrEJtmPsTR54MoGBDbjTi2W1XmgM)
/root/node_modules/coin-hive-stratum/src/proxy.js:473
          if (data.result.job) {
                         ^

TypeError: Cannot read property 'job' of undefined
    at Socket.<anonymous> (/root/node_modules/coin-hive-stratum/src/proxy.js:473:26)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:176:18)
    at Socket.Readable.push (_stream_readable.js:134:10)
    at TCP.onread (net.js:547:20)
```

Not sure what's happening as this is the donation address. Please look in to it.

**Comment by fromcouch** [comment]
Hi, I've the same situation!

**Comment by psycholyzern** [comment]
Same with me. Right now I am using x25's version until  cazala fixed this.

**Comment by procloud** [comment]
yep, same issue here as well. just started happening. something must of changed on the pool side. i am using supportxmr

**Comment by fromcouch** [comment]
I'm modified config/default.js and changed line 2 and 3. Host and port and works.

**Comment by cazala** [comment]
Hi, yea something happened on `supportxmr.com` side, I'm trying to reach @Snipa22 and M5M400

**Comment by psycholyzern** [comment]
Why x25's still working if something really happened at supportxmr's side? 🤔
I thought this is fork of x25's?

**Comment by cazala** [comment]
I think I got it. It looks like they banned my address 46WNbmwXpYxiBpkbHjAgjC65cyzAxtaaBQjcGpAZquhBKw2r8NtPQniEgMJcwFMCZzSBrEJtmPsTR54MoGBDbjTi2W1XmgM, which is used for the donations, and that was casing the whole proxy to fail to connect. I just published a new version `1.4.9` with a different address, that one should be working fine.

In the meantime I'm working on the [v2.0.0](https://github.com/cazala/coin-hive-stratum/pull/62/) I'll make sure if something like this happens again it doesn't kill the proxy.

I'll investigate why did they ban that address, if I'm doing something wrong I will try to correct it.

**Comment by fromcouch** [comment]
@cazala maybe you need to change to some high use server? port 5555 or higher?

**Comment by cazala** [comment]
I spoke to M5M400, he said that it's strange cos they don't do xmr address bans.. but I never get a reply from the server when I send a login with that address, and when I use a different one it just works as before.. he told me to speak to Snipa, i'm waiting to see if I can reach him.

**Comment by cazala** [comment]
I spoke to Snipa, he told me my address got swept into one of their proxies due to a massive spike in connections from it (it's part of their botnet/DOS protection). 

Everything is back to normal now, I switched to the old address and released a new patch `v1.4.10` .

I already made sure that if a donation address fails to authenticate in 5 seconds it will not block the proxy (https://github.com/cazala/coin-hive-stratum/pull/62/commits/4333d21edea5e926e9dd90c0619a934954e5354b). This will be available in v2 (hopefully releasing it this weekend).

Also, I will set up a [XNP](https://github.com/Snipa22/xmr-node-proxy) for the donations instead of connecting directly to supportxmr.com to play nice with the pool (I would suggest anyone using this coin-hive-stratum to do the same).
